lib/launcher.rb

```diff
@@ -1,9 +1,21 @@
+require 'bundler'
 require 'proxy/log'
 require 'proxy/sd_notify'
 require 'proxy/settings'
 require 'proxy/signal_handler'
 require 'proxy/log_buffer/trace_decorator'
 require 'thread'
+require 'rack'
+require 'webrick'
+begin
+  require 'puma'
+  require 'rack/handler/puma'
+  require 'puma_patch'
+  $HAS_PUMA = true
+rescue LoadError
+  $stderr.puts 'Puma was requested but not installed'
```

Comment:
This will always generate a warning, even when `http_server_type` is set to `webrick`

Reply:
Right, removed since we print warning few lines below.